### DISCUSSION TITLE
Creating spectral library from human oral microbiome

### DISCUSSION THREAD

**Discussion by kym-mcnicholas** [vdemichev/DiaNN#440] [Q&A]
Downloaded the entire human oral microbiome from http://www.homd.org/ as a single FASTA file. When I try to create a spectral library using this FASTA, software stops at 'processing fasta'. Suspect its an issue with memory? Any suggestions for processing a very large FASTA file. Thanks.

**Comment by vdemichev** [comment]
How does the log look like?

**Comment by kym-mcnicholas** [comment]
Thanks for your response.

I have attached log file from last run. I have tried reducing thread count (16, 8, 4) with no success – not sure if this is relevant.

Let me know if you need any more info.

Cheers,
Kym

From: Vadim Demichev ***@***.***>
Date: Tuesday, 19 July 2022 at 7:13 pm
To: vdemichev/DiaNN ***@***.***>
Cc: Kym McNicholas ***@***.***>, Author ***@***.***>
Subject: Re: [vdemichev/DiaNN] Creating spectral library from human oral microbiome (Discussion #440)
CAUTION: External email. Only click on links or open attachments from trusted senders.

________________________________
How does the log look like?

—
Reply to this email directly, view it on GitHub<https://github.com/vdemichev/DiaNN/discussions/440#discussioncomment-3179130>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZ3KRX2H26IBVTNAXDNTWE3VUZ2DTANCNFSM536V4EYQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

diann.exe --lib "" --threads 4 --verbose 1 --out "C:\DIA-NN\1.8.1\19072022oral_microbiome.tsv" --qvalue 0.01 --out-lib "C:\DIA-NN\1.8.1\19072022full_oral_microbiome-lib.tsv" --gen-spec-lib --predictor --fasta "C:\DIA-NN\1.8.1\UniPROT_Bacteria_FASTA\ALL_genomes.faa" --fasta-search --min-fr-mz 200 --max-fr-mz 1800 --cut K*,R* --missed-cleavages 1 --min-pep-len 7 --max-pep-len 30 --min-pr-mz 300 --max-pr-mz 1800 --min-pr-charge 1 --max-pr-charge 4 --unimod4 --var-mods 1 --mass-acc 10 --mass-acc-ms1 10 --reanalyse --relaxed-prot-inf --smart-profiling --peak-center --no-ifs-removal 
DIA-NN 1.8.1 (Data-Independent Acquisition by Neural Networks)
Compiled on Apr 14 2022 15:31:19
Current date and time: Tue Jul 19 15:06:10 2022
CPU: GenuineIntel Intel(R) Xeon(R) Silver 4110 CPU @ 2.10GHz
SIMD instructions: AVX AVX2 AVX512CD AVX512F FMA SSE4.1 SSE4.2 
Logical CPU cores: 32
Thread number set to 4
Output will be filtered at 0.01 FDR
A spectral library will be generated
Deep learning will be used to generate a new in silico spectral library from peptides provided
Library-free search enabled
Min fragment m/z set to 200
Max fragment m/z set to 1800
In silico digest will involve cuts at K*,R*
Maximum number of missed cleavages set to 1
Min peptide length set to 7
Max peptide length set to 30
Min precursor m/z set to 300
Max precursor m/z set to 1800
Min precursor charge set to 1
Max precursor charge set to 4
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 1
A spectral library will be created from the DIA runs and used to reanalyse them; .quant files will only be saved to disk during the first step
Highly heuristic protein grouping will be used, to reduce the number of protein groups obtained; this mode is recommended for benchmarking protein ID numbers; use with caution for anything else
When generating a spectral library, in silico predicted spectra will be retained if deemed more reliable than experimental ones
Fixed-width center of each elution peak will be used for quantification
Interference removal from fragment elution curves disabled
Mass accuracy will be fixed to 1e-05 (MS2) and 1e-05 (MS1)
Exclusion of fragments shared between heavy and light peptides from quantification is not supported in FASTA digest mode - disabled; to enable, generate an in silico predicted spectral library and analyse with this library
WARNING: MBR turned off, two or more raw files are required

0 files will be processed
[0:00] Loading FASTA C:\DIA-NN\1.8.1\UniPROT_Bacteria_FASTA\ALL_genomes.faa
[15:52] Processing FASTA

DIA-NN exited

**Comment by kym-mcnicholas** [comment]
Hi Vadim,

Just wondering if the log file was useful? Have you any suggestions for successfully creating a spectral library from this FASTA?

Thanks for your help.

Cheers,
Kym

From: Kym McNicholas ***@***.***>
Date: Tuesday, 19 July 2022 at 7:48 pm
To: vdemichev/DiaNN ***@***.***>, vdemichev/DiaNN ***@***.***>
Cc: Author ***@***.***>
Subject: Re: [vdemichev/DiaNN] Creating spectral library from human oral microbiome (Discussion #440)
Thanks for your response.

I have attached log file from last run. I have tried reducing thread count (16, 8, 4) with no success – not sure if this is relevant.

Let me know if you need any more info.

Cheers,
Kym

From: Vadim Demichev ***@***.***>
Date: Tuesday, 19 July 2022 at 7:13 pm
To: vdemichev/DiaNN ***@***.***>
Cc: Kym McNicholas ***@***.***>, Author ***@***.***>
Subject: Re: [vdemichev/DiaNN] Creating spectral library from human oral microbiome (Discussion #440)
CAUTION: External email. Only click on links or open attachments from trusted senders.

________________________________
How does the log look like?

**Reply by vdemichev** [reply]
Hi Kym, very sorry, somehow I missed the posts here. Based on the logs, looks like DIA-NN crashed because it ran out of memory. I guess the FASTA is really huge?

**Comment by kym-mcnicholas** [comment]
Thanks for your reply Vadim. Yes it's large - it's all of the bacterial proteins in the oral microbiome. Any suggestions to get around this problem?


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
